crypto/evp/p5_crpt.c

```diff
@@ -106,6 +108,7 @@ int PKCS5_PBE_keyivgen(EVP_CIPHER_CTX *cctx, const char *pass, int passlen,
     OPENSSL_cleanse(iv, EVP_MAX_IV_LENGTH);
     rv = 1;
  err:
+    PBEPARAM_free(pbe);
```

Comment:
In the successful case, the `pbe` pointer needs to be cleared after being freed in [line 86/88](https://github.com/openssl/openssl/pull/10816/files#diff-bd251e31a7360ecf59298a4344896afbR88), otherwise you'll get a double free error.

Reply:
Actually, it would be more consistent and less error prone, if you would remove the free in line 86/88 and change the `return 0` in line 91/93 to a `goto error` instead.

Reply:
Sure, `pbe` should be cleared after PBEPARAM_free in success path. I added another commit.
If just changing `goto error;` to `return 0;`, I thought that `ctx` would leak in error path.

Reply:
What I was trying to say is that IMHO it is error prone to have a mixture of `return 0` (before the first successful allocation) and  `goto error` (after the first successful allocation) error treatment inside the same function. Because this inconsistency facilitates memory leaks when adding more allocations at the beginning of the function. I'd rather 

- initialize all pointers to allocated data with NULL at the beginning of the function,
- always `goto error` on failure,
- and free all allocated data after the `error:` label.

But unfortunately, this consistent pattern is not common in the OpenSSL codebase.

Your fix is fine as it is now, no need to change anything anymore.